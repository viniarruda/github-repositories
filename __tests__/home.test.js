import React from 'react'
import { shallow } from 'enzyme'

import Home from '../src/components/views/home'

import configureStore from 'redux-mock-store'

const mockStore = configureStore()

const initialState = {
  github: {
    loading: false,
    list: null,
    error: null,
    user: null,
    commits: null
  }
}

const store = mockStore(initialState)

describe('Testing Home Component', () => {
 it('should render correctly', () => {
   const wrapper = shallow(
     <Home />,
      { context: { store } }
   ).dive();

   expect(wrapper).toMatchSnapshot()
 })
})


// Shallow = simulate DOM