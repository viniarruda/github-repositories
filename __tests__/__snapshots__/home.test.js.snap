// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    authentication={[Function]}
    github={
      Object {
        "commits": null,
        "error": null,
        "list": null,
        "loading": false,
        "user": null,
      }
    }
    searchUser={[Function]}
    sortRepos={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Loading
          show={false}
        />,
        <styled.div>
          <styled.div>
            <ForwardRef
              destroyOnUnmount={true}
              enableReinitialize={false}
              forceUnregisterOnUnmount={false}
              form="githubForm"
              getFormState={[Function]}
              keepDirtyOnReinitialize={false}
              onSubmit={[Function]}
              persistentSubmitErrors={false}
              pure={true}
              shouldAsyncValidate={[Function]}
              shouldError={[Function]}
              shouldValidate={[Function]}
              shouldWarn={[Function]}
              touchOnBlur={true}
              touchOnChange={false}
              updateUnregisteredFields={false}
            />
          </styled.div>
        </styled.div>,
        <styled.div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "show": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div>
            <ForwardRef
              destroyOnUnmount={true}
              enableReinitialize={false}
              forceUnregisterOnUnmount={false}
              form="githubForm"
              getFormState={[Function]}
              keepDirtyOnReinitialize={false}
              onSubmit={[Function]}
              persistentSubmitErrors={false}
              pure={true}
              shouldAsyncValidate={[Function]}
              shouldError={[Function]}
              shouldValidate={[Function]}
              shouldWarn={[Function]}
              touchOnBlur={true}
              touchOnChange={false}
              updateUnregisteredFields={false}
            />
          </styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ForwardRef
                destroyOnUnmount={true}
                enableReinitialize={false}
                forceUnregisterOnUnmount={false}
                form="githubForm"
                getFormState={[Function]}
                keepDirtyOnReinitialize={false}
                onSubmit={[Function]}
                persistentSubmitErrors={false}
                pure={true}
                shouldAsyncValidate={[Function]}
                shouldError={[Function]}
                shouldValidate={[Function]}
                shouldWarn={[Function]}
                touchOnBlur={true}
                touchOnChange={false}
                updateUnregisteredFields={false}
              />,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "destroyOnUnmount": true,
                "enableReinitialize": false,
                "forceUnregisterOnUnmount": false,
                "form": "githubForm",
                "getFormState": [Function],
                "keepDirtyOnReinitialize": false,
                "onSubmit": [Function],
                "persistentSubmitErrors": false,
                "pure": true,
                "shouldAsyncValidate": [Function],
                "shouldError": [Function],
                "shouldValidate": [Function],
                "shouldWarn": [Function],
                "touchOnBlur": true,
                "touchOnChange": false,
                "updateUnregisteredFields": false,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "destroyOnUnmount": true,
                  "enableReinitialize": false,
                  "forceUnregisterOnUnmount": false,
                  "form": "githubForm",
                  "getFormState": [Function],
                  "keepDirtyOnReinitialize": false,
                  "persistentSubmitErrors": false,
                  "pure": true,
                  "shouldAsyncValidate": [Function],
                  "shouldError": [Function],
                  "shouldValidate": [Function],
                  "shouldWarn": [Function],
                  "touchOnBlur": true,
                  "touchOnChange": false,
                  "updateUnregisteredFields": false,
                },
                "displayName": "ReduxForm",
                "render": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            null,
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Loading
            show={false}
          />,
          <styled.div>
            <styled.div>
              <ForwardRef
                destroyOnUnmount={true}
                enableReinitialize={false}
                forceUnregisterOnUnmount={false}
                form="githubForm"
                getFormState={[Function]}
                keepDirtyOnReinitialize={false}
                onSubmit={[Function]}
                persistentSubmitErrors={false}
                pure={true}
                shouldAsyncValidate={[Function]}
                shouldError={[Function]}
                shouldValidate={[Function]}
                shouldWarn={[Function]}
                touchOnBlur={true}
                touchOnChange={false}
                updateUnregisteredFields={false}
              />
            </styled.div>
          </styled.div>,
          <styled.div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "show": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.div>
              <ForwardRef
                destroyOnUnmount={true}
                enableReinitialize={false}
                forceUnregisterOnUnmount={false}
                form="githubForm"
                getFormState={[Function]}
                keepDirtyOnReinitialize={false}
                onSubmit={[Function]}
                persistentSubmitErrors={false}
                pure={true}
                shouldAsyncValidate={[Function]}
                shouldError={[Function]}
                shouldValidate={[Function]}
                shouldWarn={[Function]}
                touchOnBlur={true}
                touchOnChange={false}
                updateUnregisteredFields={false}
              />
            </styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <ForwardRef
                  destroyOnUnmount={true}
                  enableReinitialize={false}
                  forceUnregisterOnUnmount={false}
                  form="githubForm"
                  getFormState={[Function]}
                  keepDirtyOnReinitialize={false}
                  onSubmit={[Function]}
                  persistentSubmitErrors={false}
                  pure={true}
                  shouldAsyncValidate={[Function]}
                  shouldError={[Function]}
                  shouldValidate={[Function]}
                  shouldWarn={[Function]}
                  touchOnBlur={true}
                  touchOnChange={false}
                  updateUnregisteredFields={false}
                />,
                null,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "destroyOnUnmount": true,
                  "enableReinitialize": false,
                  "forceUnregisterOnUnmount": false,
                  "form": "githubForm",
                  "getFormState": [Function],
                  "keepDirtyOnReinitialize": false,
                  "onSubmit": [Function],
                  "persistentSubmitErrors": false,
                  "pure": true,
                  "shouldAsyncValidate": [Function],
                  "shouldError": [Function],
                  "shouldValidate": [Function],
                  "shouldWarn": [Function],
                  "touchOnBlur": true,
                  "touchOnChange": false,
                  "updateUnregisteredFields": false,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "destroyOnUnmount": true,
                    "enableReinitialize": false,
                    "forceUnregisterOnUnmount": false,
                    "form": "githubForm",
                    "getFormState": [Function],
                    "keepDirtyOnReinitialize": false,
                    "persistentSubmitErrors": false,
                    "pure": true,
                    "shouldAsyncValidate": [Function],
                    "shouldError": [Function],
                    "shouldValidate": [Function],
                    "shouldWarn": [Function],
                    "touchOnBlur": true,
                    "touchOnChange": false,
                    "updateUnregisteredFields": false,
                  },
                  "displayName": "ReduxForm",
                  "render": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              null,
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
